CONTRIBUTING.md

```diff
@@ -2,15 +2,184 @@ Guidelines for Contributing
+-   Does your PR address only one topic? Can it be split up in smaller
+    PRs?
+-   Is your code flexible enough to accommodate more use cases?
+-   Read through your code again - is there any part that isn't understandable?
+-   Try to enable `@nogc` tests
+-   Avoid code duplication
+-   Be critical when looking at Phobos code (some parts of Phobos are a
+    bit older and have some relics due to lacking functionality of
+    the compiler)
+
+### Tests
+
+-   [Autotester](https://wiki.dlang.org/Git_Commit_Tester) will automatically
+    compile the code in your PR and test it on all supported platforms.
+    For your first PR you need approval from any reviewer - ping them in
+    case they forget.
+-   Do all [tests pass locally](https://wiki.dlang.org/Starting_as_a_Contributor#Unittest_phobos)? You
+    can run the tests of
+    -   single module or packages with `make` `-f` `posix.mak`
+        `std/algorithm/comparison.test` or `make` `-f` `posix.mak`
+        `std/algorithm.test`
+    -   all tests with `make` `-f` `posix.mak` `unittest` (add `-j`
+        `NUMBER_OF_CORES` if you have multiple cores)
+    -   for small changes in a single module faster with `rdmd` `-main`
+        `-unittest` `comparison.d` (be careful, this links to your
+        current Phobos version)
+    -   for Windows have a look at [building Phobos under
+        windows](https://wiki.dlang.org/Starting_as_a_Contributor#Windows_2))
+-   Do your tests cover all cases? (you can check code coverage in the
+    resulting `module.lst` of `rdmd` `-cov` `-main` `-unittest`
+    `module.d`)
+
+Review process
+--------------
+
+-   Every PR *must* pass all tests for all supported platforms of
+    [autotester](https://wiki.dlang.org/Git_Commit_Tester) before being merged
+-   Big additions (new modules) should go to `std.experimental` first
+    (after they pass the [review process](https://wiki.dlang.org/Review/Process))
+-   Smaller additions like individual functions can be merged directly
+    after [@andralex](https://github.com/andralex) approves
+-   Refactoring or bug fixes just need approval (LGTM) from two
+    reviewers and enough time passed (usually 2-3 days, depending on
+    weekends and holidays) to give everyone an opportunity to shout
+-   For low-level changes the other compiler devs (GDC and LLVM) should
+    be consulted
+-   Trivial changes (e.g. small documentation enhancements) should be marked with
+	`[Trivial]` as title prefix
+-   If your PR is stalled (no activity within a couple of days) and you addressed
+    all concerns, don't hesitate to ping your reviewers
+-   See [rebasing](https://wiki.dlang.org/Starting_as_a_Contributor#Rebasing) if you
+    need to sync your fork with master
+
+Naming guidelines
+-----------------
+
+### General advices
+
+-   Phobos uses a distinct [naming scheme](https://wiki.dlang.org/Naming_conventions)
+-   Use a Capitalized short (50 chars or less) Git commit summary
```

Comment:
I'm not sure about this. Very often I see commits like "Fix issue 12345 - [very long issue title]" and I think that's OK. OTOH it's still a good idea in general.